crates/viewer/re_view_spatial/src/view_2d.rs

```diff
@@ -273,78 +270,55 @@ impl ViewClass for SpatialView2D {
     }
 }
 
-// Count the number of image entities with the given component exist that aren't
-// children of other entities in the bucket.
-fn count_non_nested_images_with_component(
-    image_dimensions: &IntMap<EntityPath, MaxDimensions>,
-    entity_bucket: &IntSet<EntityPath>,
-    entity_db: &re_entity_db::EntityDb,
-    subtree: &EntityTree,
-    component_name: &ComponentName,
-) -> usize {
-    if image_dimensions.contains_key(&subtree.path) {
-        // bool true -> 1
-        entity_db
-            .storage_engine()
-            .store()
-            .entity_has_component(&subtree.path, component_name) as usize
-    } else if !entity_bucket
-        .iter()
-        .any(|e| e.is_descendant_of(&subtree.path))
-    {
-        0 // early-out optimization
-    } else {
-        subtree
-            .children
-            .values()
-            .map(|child| {
-                count_non_nested_images_with_component(
-                    image_dimensions,
-                    entity_bucket,
-                    entity_db,
-                    child,
-                    component_name,
-                )
-            })
-            .sum()
-    }
+#[derive(Default, Debug)]
+struct ImageCounts {
+    image: usize,
+    encoded_image: usize,
+    depth: usize,
```

Comment:
Again, why is segmentation image missing?

Reply:
leaving a comment on this one